### DISCUSSION TITLE
Freezes when I run `next build` in scrutinizer

### DISCUSSION THREAD

**Discussion by matthewgz** [vercel/next.js#36690] [Help] [answered]
Hi, I'm trying to run the yarn build in scrutinizer, and it doesn't finish building. It stays frozen.
Does anyone know if I need any special configuration scrutinizer.

Scrutinizer stays like this, and it never ends:

![scrutinizer-ci com_g_matthewgz_test_inspections_5451868a-a11d-49ac-be63-a5556dc9e840](https://user-images.githubusercontent.com/21980013/166839336-e37f43a9-f1e5-4c2a-95ac-a438642fc133.png)

Repository with which the error occurs: https://github.com/matthewgz/test

_Note: The build only fails on scrutinizer, on my localhost it works fine._

Thank you in advance.

**Comment by icyJoseph** [comment]
I remember having an issue like this, in Vercel, where my build was timing out, and it was indeed stuck at checking validity of types. 

After a bit of research I figured out I had made some recursive type that never ended, but it only happened with certain version of styled-components types. 

Either, the build step lacks memory, or there's some type check that never ends. Locally, try to clean your lock files, install everything again, make sure you are using the same node version as your deployed project, and see if you can get it to hang locally as well.

What Next.js version do you use? and which Node.js version? Do you see any logs or the like from Scrutinizer?

**Reply by matthewgz** [reply]
I am using the following versions:
NextJS - 12.1.6
Node-16.15.0

The only log I have gives me Scrutinizer is this:

![image](https://user-images.githubusercontent.com/21980013/166939618-2b931a3d-31c7-4ad5-a608-08a512048cac.png)

The repository I'm using is the default template that nextjs gives you on the first install, I didn't add anything new.

Another thing I discovered is that this bug only happens with versions next@12 onwards, but with version next@11.1.4 the build is done correctly in scrutinizer 🤔

_Note: I already deleted node_modules, .next, yarn.lock and reinstalled everything and the same thing happens in Scrutinizer._

**Reply by icyJoseph** [reply]
Alright, but that message is revealing. I wonder if there's something that `swc` is not interpreting correctly.

A bit dirty, but if you increase the `idle_timeout` to 10 seconds (if possible), perhaps the build will go on.

At this point, I imagine the problem ought to be something with SWC, have you checked other discussions regarding this? Maybe take a look here https://github.com/vercel/next.js/discussions/34961.

Though, you say you haven't changed anything on the starter template. Very odd. Should we think that perhaps, SWC can't compile?

**Reply by matthewgz** [reply]
I tried setting 10000 seconds, but scrutinizer aborted the build at 7200 seconds (2h). I really don't know what it could be.

![image](https://user-images.githubusercontent.com/21980013/167058546-714ee7d2-e142-4df2-827f-4d2e646b4e72.png)

**Reply by icyJoseph** [reply]
I tried doing some googling, and this was the first result :D - I think someone else with Scrutinizer experience has to take a look at this, but again, I'd think that the problem is with SWC not being supported, which is the one of the major differences in Next 12.

**Comment by matthewgz** [comment, marked as answer]
**UPDATE**

I had support from scrutinizer, and it seems that the version of ubuntu used by the scrutinizer container by default is incompatible (Ubuntu 14.04). So that is fixed by upgrading to version 18.04 adding this line `image: default-bionic`

![image](https://user-images.githubusercontent.com/21980013/167661273-9f229e00-019a-4e13-9e46-ea2de2330301.png)

And now everything compiles perfectly

![image](https://user-images.githubusercontent.com/21980013/167661397-b0d40772-ddf9-4f06-8bcd-0dc11f87b9e2.png)

I hope it helps someone else 😁

**Reply by icyJoseph** [reply]
Awesome! this was really tricky, good job.
